### PR TITLE
Add first layer of tests

### PR DESCRIPTION
**Pull request by ponsoc**
Adds the first tests but skips the property getter and setters for now since they are harder to test. Will be done in a separate PR

### CODE REVIEW

File: tests/component.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+def component(mock_store):
+    # Creating a sample component for testing
+    component = Component(component_id="test_component", initial_state={"key": "value"})
+    component._Component__store = mock_store  # Injecting the mock store into the component
+    return component
+
+def test_component_initialization(component):
+    # Test that the component is initialized correctly
+    assert component.id == "test_component"
+    assert component.state is not None
```
Comment (justinkraaijenbrink) [inline tests/component.py:22]: state zou een vaste vorm moeten zijn (dict?), dus misschien beter daarop testen?

File: tests/router.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import pytest
```
Comment (justinkraaijenbrink) [inline tests/router.py:1]: misschien nog goed om te kijken naar het testen van edge cases, bv als een invalid route wordt aangeroepen oid

**Review by justinkraaijenbrink — COMMENTED** [review]
testing is niet mijn specialiteit, hopelijk heb je er wat aan 😅

File: tests/router.py (hunk 2 above)
Comment (ponsoc) [reply tests/router.py:1]: Goed punt heb error handeling toegevoegd en hier een test op gemaakt.
